net/client/limitParallelRequests/limitParallelRequests.go

```diff
@@ -0,0 +1,135 @@
+package limitparallelrequests
+
+import (
+	"context"
+	"fmt"
+	"hash/crc64"
+	"math"
+
+	"github.com/plgd-dev/go-coap/v3/message"
+	"github.com/plgd-dev/go-coap/v3/message/pool"
+	coapSync "github.com/plgd-dev/go-coap/v3/pkg/sync"
+	"golang.org/x/sync/semaphore"
+)
+
+type (
+	DoFunc        = func(req *pool.Message) (*pool.Message, error)
+	DoObserveFunc = func(req *pool.Message, observeFunc func(req *pool.Message), opts ...message.Option) (Observation, error)
```

Comment:
DoFunc and DoObserveFunc are already defined in [net/client/client.go], seems duplicit.

Reply:
Removed from `net/client/client.go`